### PR TITLE
fix(discord): Fix log messages for Discord

### PR DESCRIPTION
**Pull request by 17hogeju**
Fixes log messages and metric details for Discord such that discord_code and message can be logged.

`if resp` returns false even when there is a response and mypy doesn't understand so I added a type ignore.

### CODE REVIEW

File: src/sentry/integrations/discord/actions/issue_alert/notification.py [hunk 1]
```diff
@@ -49,7 +48,7 @@ def send_notification(event: GroupEvent, futures: Sequence[RuleFuture]) -> None:
             client = DiscordClient()
             try:
                 client.send_message(channel_id, message, notification_uuid=notification_uuid)
-            except ApiError as e:
+            except Exception as e:
```
Comment (17hogeju) [inline src/sentry/integrations/discord/actions/issue_alert/notification.py:51]: This extra info was not being logged since the error was not ApiError

Comment (17hogeju) [reply src/sentry/integrations/discord/actions/issue_alert/notification.py:51]: Same with the send_message in metric_alert.py

Comment (ykamo001) [reply src/sentry/integrations/discord/actions/issue_alert/notification.py:51]: The base client class has some control over the error type, which is why we might not have been catching all errors: https://github.com/getsentry/sentry/blob/bb1a2ef98d0b9d4b5fb77c38daa8511bf1b20596/src/sentry/shared_integrations/client/base.py#L252-L300

**Review by ykamo001 — APPROVED** [review]

File: src/sentry/integrations/discord/actions/metric_alert.py [hunk 2]
```diff
@@ -53,7 +52,7 @@ def send_incident_alert_notification(
     client = DiscordClient()
     try:
         client.send_message(channel, message)
-    except ApiError as error:
+    except Exception as error:
         logger.warning(
```
Comment (ceorourke) [inline src/sentry/integrations/discord/actions/metric_alert.py:56]: nit while you're in here - `messsage_send_failure`

File: src/sentry/integrations/discord/client.py [hunk 3]
```diff
@@ -117,15 +117,19 @@ def track_response_data(
...
+        """
         discord_code = ""
         code_message = ""
 
         try:
-            discord_response = json.loads(resp.content.decode("utf-8")) if resp else {}
-            discord_code = discord_response.get("code")
-            if discord_code in DISCORD_ERROR_CODES:
+            discord_response: dict = json.loads(resp.content.decode("utf-8")) or {}  # type: ignore
+            discord_code = discord_response.get("code") or ""
```
Comment (ceorourke) [inline src/sentry/integrations/discord/client.py:130]: You can put the empty string within the get like `.get("code", "")`

**Pull request merged**